### PR TITLE
impl ExportableProvider for ForkByErrorProvider and add tutorial

### PR DESCRIPTION
**Pull request by sffc**
@jedel1043

### CODE REVIEW

**Review by Manishearth — DISMISSED** [review]

**Review by Manishearth — APPROVED** [review]

File: provider/adapters/Cargo.toml [hunk 1]
```diff
@@ -29,3 +29,7 @@ serde = { workspace = true, features = ["derive", "alloc"], optional = true }
 icu_provider = { path = "../../provider/core", features = ["macros", "deserialize_json"] }
 icu_locale = { path = "../../components/locale" }
 writeable = { path = "../../utils/writeable" }
+
+[features]
+std = []
+export = ["icu_provider/export", "std"]
```
Comment (robertbastian) [inline provider/adapters/Cargo.toml:35]: It seems incomplete to not implement `ExportableProvider` on all adapter types.

Comment (robertbastian) [reply provider/adapters/Cargo.toml:35]: `ForkByMarkerProvider` and `MultiForkByMarkerProvider` are still missing.

File: provider/adapters/src/fork/by_error.rs [hunk 2]
```diff
@@ -334,3 +350,18 @@ where
...
+where
+    P: ExportableProvider,
+    F: ForkByErrorPredicate + Sync,
+{
+    fn supported_markers(&self) -> std::collections::HashSet<DataMarkerInfo> {
+        let mut markers = std::collections::HashSet::new();
+        for provider in self.providers.iter() {
+            markers.extend(provider.supported_markers());
+        }
+        markers
```
Comment (robertbastian) [inline provider/adapters/src/fork/by_error.rs:365]:
```suggestion
        self.providers.iter().flat_map(|p| p.supported_markers()).collect()
```

File: provider/adapters/src/fork/by_error.rs [hunk 3]
```diff
@@ -159,6 +161,20 @@ where
...
+impl<P0, P1, F> ExportableProvider for ForkByErrorProvider<P0, P1, F>
+where
+    P0: ExportableProvider,
+    P1: ExportableProvider,
+    F: ForkByErrorPredicate + Sync,
+{
+    fn supported_markers(&self) -> std::collections::HashSet<DataMarkerInfo> {
+        let mut markers = self.0.supported_markers();
+        markers.extend(self.1.supported_markers());
+        markers
```
Comment (robertbastian) [inline provider/adapters/src/fork/by_error.rs:174]:
```suggestion
        [&self.0, &self.1].iter().map(|p| p.supported_markers()).collect()
```

**Review by Manishearth — DISMISSED** [review]

File: provider/adapters/src/fork/by_error.rs (hunk 3 above)
Comment (sffc) [reply provider/adapters/src/fork/by_error.rs:174]: Not sure

File: provider/adapters/Cargo.toml (hunk 1 above)
Comment (sffc) [reply provider/adapters/Cargo.toml:35]:
```
pub type ForkByMarkerProvider<P0, P1> = ForkByErrorProvider<P0, P1, MarkerNotFoundPredicate>;
```

**Review by robertbastian — DISMISSED** [review]

**Review by robertbastian — APPROVED** [review]

**Pull request merged**